std/src/main/scala/cats/std/map.scala

```diff
@@ -1,31 +1,57 @@
 package cats
 package std
 
-
 trait MapInstances extends algebra.std.MapInstances {
 
   implicit def MapShow[A, B](implicit showA: Show[A], showB: Show[B]): Show[Map[A, B]] =
-    Show.show[Map[A, B]](m => s"Map(${m.map(a => s"${showA.show(a._1)} -> ${showB.show(a._2)})").mkString(",")})")
+    Show.show[Map[A, B]] { m =>
+      val body = m.map { case (a, b) =>
+        s"${showA.show(a)} -> ${showB.show(b)})"
+      }.mkString(",")
+      s"Map($body)"
+    }
 
   implicit def mapInstance[K]: Traverse[Map[K, ?]] with FlatMap[Map[K, ?]] =
-    new Traverse[Map[K, ?]] with FlatMap[Map[K, ?]] {
-      def traverse[G[_] : Applicative, A, B](fa: Map[K, A])(f: (A) => G[B]): G[Map[K, B]] = {
-        val G = Applicative[G]
-        val gba = G.pure(Map.empty[K, B])
-        val gbb = fa.foldLeft(gba)((buf, a) => G.map2(buf, f(a._2))({ case(x, y) => x + (a._1 -> y)}))
-        G.map(gbb)(_.toMap)
-      }
+    new MapInstance[K]
```

Comment:
My 2 cents —  when I am trying to look up an instance's definition and I run into a proper class instead of an anonymous class, I feel like it's an additional layer of indirection between me and the definition.